lib/slack-rtmapi/client.rb

```diff
@@ -81,14 +81,22 @@ def connected?
       @connected || false
     end
 
+    def read_socket
+      @reading = true
+      data = @socket.readpartial 4096
+      @driver.parse data if !data.nil? && !data.empty?
+      @reading = false
+    end
+
     # All the polling work is done here
     def inner_loop
       return if @stop
-      data = @socket.readpartial 4096
-      return if data.nil? or data.empty?
-
-      @driver.parse data
-      @msg_queue.each {|msg| @driver.text msg}
+      unless @reading
+        Thread.new do
```

Comment:
The flip side of this is that no communication can happen unless data comes in...